### ISSUE TITLE
Expose HTLCs in `ChannelDetails`

### ISSUE THREAD

**Issue by TheBlueMatt** [lightningdevkit/rust-lightning#2087]
We should expose the list of in-flight HTLCs in `ChannelDetails`, cause why not. This should let us better figure out what the timeline is for the channel close. We should also expose if a channel is actively shutting down and maybe if a peer is currently connected in `ChannelDetails`. May help with visibility into #2086.

**Comment by ayeankit** [comment]
hey @TheBlueMatt, Is this still open? i would like to work on this.

**Comment by aniket2405** [comment]
hey, it seems that the PR for this was not accepted and closed. Is this still open then? If yes, I would like to work on this.

**Comment by aniket2405** [comment]
Do these fields in the `ChannelDetails` struct in the `ln::channelmanager` module make sense to expose the HTLCs? The first is present already, what about the rest?

pub struct `ChannelDetails` {
    pub `channel_id`: [u8; 32],
    pub `local_balance`: u64,
    pub `remote_balance`: u64,
    pub `remote_pubkey`: PublicKey,
    pub `channel_value_satoshis`: u64,
    pub `user_id`: UserId,
    pub `in_flight_htlcs`: Vec<(PaymentHash, u64, HTLCType)>,
    pub `is_shutting_down`: bool,
    pub `peer_connected`: bool,
}

**Comment by tlulu** [comment]
Is anyone actively working on this? If not, I would like to work on this. For context, I want to expose an HTLC's cltv_expiry. 

Alternatively would an API to query LDK be better? E.g 
```
fn get_htlc(payment_hash: PaymentHash) -> HtlcDetails

pub struct HtlcDetails {
  pub absolute_cltv_expiry: Duration
  pub payment_hash: PaymentHash
}
```

**Comment by wvanlint** [comment]
Yes, sent out https://github.com/lightningdevkit/rust-lightning/pull/2442 for review.

**Comment by malsomesh9** [comment]
hey @TheBlueMatt, Is this still open? i would like to work on this

**Comment by TheBlueMatt** [comment]
Oops, no, this was fixed in #2442.
